benchmark/MainPrintTimes.cpp

```diff
@@ -0,0 +1,164 @@
+  const size_t exactNumberOfDecimals = 2;
+
+  // How long is the string of a float, that has exactNumberOfDecimals amount
+  // of decimals?
+  auto stringLenghtOfFloat = [&exactNumberOfDecimals](const float number) {
+    std::stringstream ss;
+    ss << std::fixed << std::setprecision(exactNumberOfDecimals) << number;
+    return ss.str().length();
+  };
+
+  // Add a string to a stringstream with enough padding, empty spaces to the
+  // right of the string, to reach the wanted length.
+  auto addStringWithPadding = [](std::stringstream* stringStream,
+      const auto& text, const size_t wantedLength){
+    (*stringStream) << std::setw(wantedLength) << std::left << text;
+  };
+
+  // Printing the tables themselves.
+  for (const auto& entry: records.getTables()) {
+    const BenchmarkRecords::RecordTable& table = entry.second;
+    visualization << "\n\nTable '" << table.descriptor << "':\n\n";
+  
+    // For easier usage.
+    size_t numberColumns = table.columnNames.size();
+    size_t numberRows = table.rowNames.size();
+
+    // For formating: What is the maximum string width of a column, if you
+    // compare all it's entries?
+    const size_t rowNameMaxStringWidth = std::ranges::max(table.rowNames,
+        {}, [](const std::string& name){return name.length();}).length();
+
+    std::vector<size_t> columnMaxStringWidth(numberColumns, 0);
+    for (size_t column = 0; column < numberColumns; column++) {
+      // Which of the entries is the longest?
+      columnMaxStringWidth[column] = stringLenghtOfFloat(
+          std::ranges::max(table.entries, {},
+          [&column, &stringLenghtOfFloat](const std::vector<float>& row){
+          return stringLenghtOfFloat(row[column]);})[column]);
+      // Is the name of the column bigger?
+      columnMaxStringWidth[column] =
+        (columnMaxStringWidth[column] > table.columnNames[column].length()) ?
+        columnMaxStringWidth[column] : table.columnNames[column].length();
+    }
+
+    // Because the column of row names also has a width, we create an empty
+    // string of that size, before actually printing the row of column names.
+    visualization << std::string(rowNameMaxStringWidth, ' ');
+
+    // Print the top row of names.
+    for (size_t column = 0; column < numberColumns; column++){
+      visualization << "\t";
+      addStringWithPadding(&visualization, table.columnNames[column], columnMaxStringWidth[column]);
+    }
+
+    // Print the rows.
+    for (size_t row = 0; row < numberRows; row++) {
+      // Row name
+      visualization << "\n";
+      addStringWithPadding(&visualization, table.rowNames[row], rowNameMaxStringWidth);
+      visualization << "\t";
+
+      // Row content.
+      for (size_t column = 0; column < numberColumns; column++) {
+        addStringWithPadding(&visualization, table.entries[row][column], columnMaxStringWidth[column]);
+        visualization << "\t";
+      }
+    }
+  }
+
+  std::cout << visualization.str() << "\n";
+};
```

Comment:
I am still not sure whether we need the three main functions.
I think we should better configure this in the benchmarks (e.g. often a table with two columns also implicitly has a speedup. And averages also have to be configured (e.g. average over different inputs of the same size land in the same measurement name etc.
But we best discuss this in person.